aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1308 @@
+| [Environment variables Configuration Provider](#environment-variables-configuration-provider) | Environment variables |
+| [File Configuration Provider](#file-configuration-provider) | Files (INI, JSON, XML) |
+| [Key-per-file Configuration Provider](#key-per-file-configuration-provider) | Directory files |
+| [Memory Configuration Provider](#memory-configuration-provider) | In-memory collections |
 
 ::: moniker-end
 
-::: moniker range="<= aspnetcore-1.1"
+::: moniker range="< aspnetcore-2.1"
+
+| Provider | Provides configuration from&hellip; |
+| -------- | ----------------------------------- |
+| [Command-line Configuration Provider](#command-line-configuration-provider) | Command-line parameters |
+| [Environment variables Configuration Provider](#environment-variables-configuration-provider) | Environment variables |
+| [File Configuration Provider](#file-configuration-provider) | Files (INI, JSON, XML) |
+| [Memory Configuration Provider](#memory-configuration-provider) | In-memory collections |
+
+::: moniker-end
 
-Examples provided in this topic rely upon:
+::: moniker range=">= aspnetcore-1.1"
 
-* Setting the base path of the app with [SetBasePath](/dotnet/api/microsoft.extensions.configuration.fileconfigurationextensions.setbasepath). `SetBasePath` is made available to the app by referencing the [Microsoft.Extensions.Configuration.FileExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.FileExtensions/) package.
-* Resolving sections of configuration files with [GetSection](/dotnet/api/microsoft.extensions.configuration.configurationsection.getsection). `GetSection` is made available to the app by referencing the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package.
-* Binding configuration with [Bind](/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind). `Bind` is made available to the app by referencing the [Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) package.
+Additional providers described in other topics are shown in the following table.
+
+| Provider | Provides configuration from&hellip; |
+| -------- | ----------------------------------- |
+| [Azure Key Vault Configuration Provider](xref:security/key-vault-configuration) | Azure Key Vault |
 
 ::: moniker-end
 
-## JSON configuration
+You can also create a [Custom configuration provider](#custom-configuration-provider).
+
+Except for using a File Configuration Provider's reload-on-change capability, configuration sources are read in the order that their configuration providers are specified at startup. The configuration providers described in this topic are described in alphabetical order, not in the order that your code may arrange them. Order configuration providers in your code to suit your priorities for the underlying configuration sources.
```

Comment:
I'm describing reload-on-change, not `IConfigurationRoot.Reload`. Doesn't the change token trip the provider (and only that provider) to read the source? Since only that source is read (if that's correct), then it's fair to say that config sources are read in the order provided *except for reload-on-change*, which circumvents the initial order and priority. This might be an incorrect view, since I'm not digging thru source code (yet) to see. It's just the received view from looking at the API comments.

Reply:
When the change token tripps the entire config is rebuilt in the original order. A reloaded config file won't overwrite values from a higher priority provider.

Reply:
Ok thanks ... I'm a bit surprised that that's the design.

I'll fix it on the next commit.